### ISSUE TITLE
testcompile dependencies are not part of generated pom.xml from nebula maven publication

### ISSUE THREAD

**Issue by kalaiarasan231** [nebula-plugins/nebula-publishing-plugin#171]
Apart from scope test, dependencies from compile and runtime are properly coming in generated pom.xml.
Is there a way to include test dependencies as well? For some reason, have to use gradle 4.10.2, knowingly it is old now.
can some one help me on this? 

apply plugin: 'nebula.maven-publish'

group="gradle-lib-nexus"
version="2.0"

dependencies {
    // This dependency is exported to consumers, that is to say found on their compile classpath.
    compile 'org.apache.commons:commons-math3:3.+'

    // This dependency is used internally, and not exposed to consumers on their own compile classpath.
    compile 'com.google.guava:guava:22.+'  // This looks good along with above one.

    // Use JUnit test framework
     testCompile 'junit:junit:4.12'                 // **Not part of generated POM**
}

**Deployment as:** 
/* the below  is for publishing individual artifacts */
publishing {
  repositories {
    maven {
      credentials {
        username "$nexusUser"
        password "$nexusPassword"
    }
      url "repo"
    }
  }
}

**POM contains:**   ( Need Junit dependency also to be added as scope test - is there a way ? ) 
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
<modelVersion>4.0.0</modelVersion>
<groupId>gradle-lib-nexus</groupId>
<artifactId>gradle-lib-nexus</artifactId>
<version>2.0</version>
<dependencies>
<dependency>
<groupId>com.google.guava</groupId>
<artifactId>guava</artifactId>
<version>22.0</version>
<scope>compile</scope>
</dependency>
<dependency>
<groupId>org.apache.commons</groupId>
<artifactId>commons-math3</artifactId>
<version>3.6.1</version>
<scope>compile</scope>
</dependency>
</dependencies>
<name>gradle-lib-nexus</name>
<description/>
</project>

**Comment by rpalcolea** [comment]
Hi,

I don't think we have ergonomics in the plugin to publish those automatically. We usually recommend folks to publish common test libraries as another module or use [Gradle Test Fixtures](https://docs.gradle.org/current/userguide/java_testing.html#sec:java_test_fixtures). More info in https://docs.gradle.org/current/userguide/java_testing.html#publishing_test_fixtures.

However, test fixtures are only available in Gradle 5.6+

We recommend that because the POM really only needs compile/runtime dependencies. Fixtures seem like a better fit for your use case

One thing that you could do is modify the XML yourself via gradle api and read the dependencies from the test configuration and attach them, but that would make your POM file to contain the test dependencies in there, so you have to make sure you add them with the `test` scope to avoid consumers pulling those automatically

I guess you could do something like this:

```
project.afterEvaluate {
    project.publishing {
        publications {
            withType(MavenPublication) {
                pom.withXml { XmlProvider xml ->
                    project.plugins.withType(JavaBasePlugin) {
                        def dependencies = xml.asNode()?.dependencies?.dependency
                        project.configurations.test.resolvedConfiguration.firstLevelModuleDependencies.each { dep ->
                            Node dependency = new Node(null, 'dependency')
                            dependency.appendNode('groupId', dep.moduleGroup)
                            dependency.appendNode('artifactId', dep.moduleName)
                            dependency.appendNode('version', dep.moduleVersion)
                            dependency.appendNode('scope', 'test')
                            dependencies.append(dependency)
                        }
                    }
                }
            }
        }
    }
}
``` 

I don't know if that will work but you can give it a shot. Notice that this will add all dependencies in the configuration... so this could be a huge number of them

Again, I'd recommend upgrade and start using [Gradle Test Fixtures](https://docs.gradle.org/current/userguide/java_testing.html#sec:java_test_fixtures)

I'll close this issue as there is nothing we can do on nebula publishing plugin side for this

**Comment by kalaiarasan231** [comment]
Thank you @rpalcolea . Your kind information helped me to go forward. will look at Gradle Test Fixtures for neat solution.
